File: python/leetcode/249.py
class Solution(object):
    def groupStrings(self, strings):
        d = collections.defaultdict(list)
        strings.sort()
        for word in strings:
            shift = [chr((ord(ch) - ord(word[0])) % 26) for ch in word]
            shift = "".join(shift) if shift else " "
            d[shift].append(word)
        return d.values()

File: cpp/leetcode/249.cpp
class Solution {
public:
    vector<vector<string>> groupStrings(vector<string>& strings) {
        unordered_map<string, vector<string>> map;
        vector<vector<string>> res;
        for(auto strs: strings){
            string key = "";
            for(int i = 0; i < strs.size(); i++){
                key += char((strs[i] - strs[0] + 26)%26 + 'a');
            }
            map[key].push_back(strs);
        }
        for(auto item:map) res.push_back(item.second);
        return res;
    }
};